File: tests/fixtures/add-python-clojure-julia-frontends/5/runner-input.py
"""Python source for runner-input extraction testing.

Expected runner-input fields:
- version: "0.1.0"
- source_file: "runner-input.py"
- tagged_fns: functions with type annotations
- serializable_values: parameters with primitive types
- generator_refs: generator functions (yield)
"""

from typing import Optional


def add(a: int, b: int) -> int:
    return a + b


def greet(name: str) -> str:
    return f"Hello, {name}!"


def process(items: list[str]) -> Optional[str]:
    if items:
        return items[0]
    return None


def count_up_to(n: int):
    """Generator function."""
    for i in range(n):
        yield i

File: tests/fixtures/add-python-clojure-julia-frontends/6/runner-input.clj
;; Clojure source for runner-input extraction testing.
;;
;; Expected runner-input fields:
;; - version: "0.1.0"
;; - source_file: "runner-input.clj"
;; - tagged_fns: defn declarations with params and return types
;; - serializable_values: parameters with primitive types
;; - generator_refs: lazy-seq generators

(ns runner-input.core)

(defn add
  "Add two numbers."
  [a b]
  (+ a b))

(defn greet
  "Greet someone."
  [name]
  (str "Hello, " name "!"))

(defn process
  "Process a collection."
  [items]
  (when (seq items)
    (first items)))

(defn count-up-to
  "Generator function."
  [n]
  (lazy-seq (cons n (count-up-to (inc n)))))